app/celery/process_delivery_status_result_tasks.py

```diff
@@ -85,11 +84,11 @@ def process_delivery_status(
     try:
         # calculate pricing
         current_app.logger.info(
-            'Notification ID (%s) - Calculate Pricing: %s and notification_status: %s with number_of_message_parts: %s',
-            notification.id,
+            'Calculate Pricing: %s and update notification_status: %s with number_of_message_parts: %s for notification: %s',
             provider_name,
             notification_status,
             number_of_message_parts,
+            notification.id,
```

Comment:
If the notification is None. It should not make it here, it should return at the above statement. I'll add it though, since we wouldn't want to retry in that case.